tests/contrib/pytest/test_pytest.py

```diff
@@ -123,18 +145,14 @@ def item_param():
                 return 42
 
             @pytest.mark.parametrize(
-                'item',
-                [
-                    1,
-                    2,
-                    3,
-                    4,
-                    pytest.param(A("test_name", "value"), marks=pytest.mark.skip),
-                    pytest.param(A("test_name", A("inner_name", "value")), marks=pytest.mark.skip),
-                    pytest.param({"a": A("test_name", "value"), "b": [1, 2, 3]}, marks=pytest.mark.skip),
-                    pytest.param([1, 2, 3], marks=pytest.mark.skip),
-                    pytest.param(item_param, marks=pytest.mark.skip)
-                ]
+            'item',
+            [
+                pytest.param(A("test_name", "value"), marks=pytest.mark.skip),
+                pytest.param(A("test_name", A("inner_name", "value")), marks=pytest.mark.skip),
+                pytest.param(item_param, marks=pytest.mark.skip),
+                pytest.param({"a": A("test_name", "value"), "b": [1, 2, 3]}, marks=pytest.mark.skip),
+                pytest.param(MagicMock(value=MagicMock()), marks=pytest.mark.skip),
```

Comment:
If this is the case added to reproduce a circular reference - can we be more explicit. MagicMock could change its internals at any time which might remove the circular reference. Can we create our own simple class that contains a circular reference instead?

Reply:
That does not seem circular to me.
I think we miss such a case?

Reply:
Yes, `MagicMock` objects that contain `MagicMock` attributes result in a circular reference between the child and the parent, which was the cause of this issue. I've added in a custom simple case of a circular reference as well 👍

Reply:
Oh ok! This was not obvious indeed.
Thank you!